config/crd/bases/ironic.openstack.org_ironicneutronagents.yaml

```diff
@@ -0,0 +1,232 @@
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: IronicNeutronAgentSpec defines the desired state of ML2 baremetal
+              - ironic-neutron-agent agents
+            properties:
+              containerImage:
+                description: ContainerImage - ML2 baremtal - Ironic Neutron Agent
+                  Image URL
+                type: string
+              customServiceConfig:
+                default: '# add your customization here'
+                description: CustomServiceConfig - customize the service config using
+                  this parameter to change service defaults, or overwrite rendered
+                  information using raw OpenStack config format. The content gets
+                  added to to /etc/<service>/<service>.conf.d directory as custom.conf
+                  file.
+                type: string
+              debug:
+                description: Debug - enable debug for different deploy stages. If
+                  an init container is used, it runs and the actual action pod gets
+                  started with sleep infinity
+                properties:
+                  bootstrap:
+                    default: false
+                    description: ReadyCount enable debug
+                    type: boolean
+                  dbSync:
+                    default: false
+                    description: DBSync enable debug
+                    type: boolean
+                  service:
+                    default: false
+                    description: Service enable debug
+                    type: boolean
+                type: object
+              networkAttachmentDefinitions:
+                description: NetworkAttachmentDefinitions list of network attachment
+                  definitions the service pod gets attached to
+                items:
+                  type: string
+                type: array
+              nodeSelector:
+                additionalProperties:
+                  type: string
+                description: NodeSelector to target subset of worker nodes running
+                  this service.
+                type: object
+              passwordSelectors:
+                default:
+                  service: IronicPassword
+                description: PasswordSelectors - Selectors to identify the ServiceUser
+                  password and TransportURL from the Secret
+                properties:
+                  database:
+                    default: IronicDatabasePassword
```

Comment:
... Why does this part of the CRD need a password?

Reply:
The CRD is generated from code, I re-used PasswordSelectors from ironic_types.go. (https://github.com/openstack-k8s-operators/ironic-operator/blob/master/api/v1beta1/ironic_types.go#L118)
Updated, there is a separate IronicNeutronAgentPasswordSelector with just the service now.